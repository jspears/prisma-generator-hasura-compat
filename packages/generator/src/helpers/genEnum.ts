import { DMMF } from '@prisma/generator-helper'
import type {  DocumentNode,  EnumTypeDefinitionNode,
  DefinitionNode } from 'graphql'
import { print,Kind } from 'graphql';
type EnumType = DMMF.DatamodelEnum;
type EnumValue = DMMF.EnumValue;


export function generateAllEnums(enums:EnumType[]):DocumentNode{

  const definitions:DefinitionNode[] = enums.map(({ name,dbName, values }) =>({
    kind: Kind.ENUM_TYPE_DEFINITION,
    name: { kind: Kind.NAME, value:  dbName ?? name },
    values: values.map(v=>({
      kind: Kind.ENUM_VALUE_DEFINITION,
      name: { kind: Kind.NAME, value: v.dbName ??v.name },
    }))
  }) )

  return {
    kind: Kind.DOCUMENT,
    definitions
  } as const;
}

export function printAllEnums(enums:EnumType[]){
  return print(generateAllEnums(enums));
}

