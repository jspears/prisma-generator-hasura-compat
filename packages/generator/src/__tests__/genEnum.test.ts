import {  generateAllEnums } from '../helpers/genEnum'
import { getSampleDMMF } from './__fixtures__/getSampleDMMF'
import {print} from 'graphql';

test('enum generation', async () => {
  const sampleDMMF = await getSampleDMMF()

  expect(print(generateAllEnums(sampleDMMF.datamodel.enums))).toMatchSnapshot('enums')
})
