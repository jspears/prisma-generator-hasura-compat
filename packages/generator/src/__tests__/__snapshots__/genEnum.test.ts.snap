// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enum generation: enums 1`] = `
"enum notification_type {
  new_posts
  new_comments
  newFollowers
  reply
  heartOnPost
  heartOnComment
  heartOnReply
}

enum Language {
  Typescript
  Javascript
  Rust
  Go
  Python
  Cpp
}

enum State {
  Active
  Pending
  Banned
}"
`;
